lib/fluent/plugin/in_systemd.rb

```diff
@@ -55,7 +56,11 @@ def read_from
 
     def run
       watch do |entry|
-        router.emit(tag, entry.realtime_timestamp.to_i, formatted(entry))
+        begin
+          router.emit(tag, entry.realtime_timestamp.to_i, formatted(entry))
+        rescue => e
+          log.error("Exception emitting record: #{e}")
+        end
```

Comment:
There's definitely a bug, but it's not in this plugin. In my case the kubernetes client that the kubernetes metadata filter plugin uses sometimes throws an SSL error which isn't wrapped properly. This kills poor journald daemon thread 😔. My plan is to report it upstream once I save the exact exception.